sale_loyalty_order_suggestion/wizard/sale_loyalty_reward_wizard.py

```diff
@@ -0,0 +1,172 @@
+                    lambda x, record=record: x.product_id == record
+                    and not x.is_reward_line
+                ).product_uom_qty
+                lines_vals.append(
+                    (
+                        0,
+                        0,
+                        {
+                            "wizard_id": self.id,
+                            "product_id": record.id,
+                            "units_required": units_required,
+                            "units_included": units_included or 0,
+                        },
+                    )
+                )
+            self.loyalty_rule_line_ids = lines_vals
+
+    @api.depends_context("lang")
+    @api.depends("selected_reward_id")
+    def _compute_loyalty_rule_line_description(self):
+        self.loyalty_rule_line_description = False
+        units_required = min(
+            self.selected_reward_id.program_id.rule_ids.mapped("minimum_qty"), default=0
+        )
+        if (
+            self.selected_reward_id
+            and not self.applicable_program
+            and units_required > 0
+        ):
+            products = self.loyalty_rule_line_ids.mapped("product_id.name")
+            if len(products) > 1:
+                products_str = f"{', '.join(products[:-1])} {_('or')} {products[-1]}"
+            else:
+                products_str = products[0] if products else ""
+            self.loyalty_rule_line_description = _(
+                "<b>* Required quantity:</b> %(units)s units of %(products)s"
+            ) % {
+                "units": units_required,
+                "products": products_str,
+            }
+
+    def _update_order_line_with_units(self, order_line, units):
+        """Updates an existing order line with the provided units."""
+        order_line.write({"product_uom_qty": order_line.product_uom_qty + units})
+
+    def _create_new_order_line(self, product, units):
+        """Creates a new order line with the product and units provided."""
+        self.order_id.order_line.create(
+            {
+                "order_id": self.order_id.id,
+                "product_id": product.id,
+                "product_uom_qty": units,
+            }
+        )
+
+    def action_apply(self):
+        for line in self.loyalty_rule_line_ids.filtered(
+            lambda x: x.units_to_include > 0
+        ):
+            # Filter existing order lines based on the product of the loyalty rule line
+            order_line = self.order_id.order_line.filtered(
+                lambda x, line=line: x.product_id == line.product_id
+            )
+            if order_line:
+                self._update_order_line_with_units(order_line, line.units_to_include)
+            else:
+                self._create_new_order_line(line.product_id, line.units_to_include)
+        self.order_id._update_programs_and_rewards()
+        super().action_apply()
+        return {
+            "type": "ir.actions.client",
+            "tag": "reload",
```

Comment:
Could you please try this? To prevent reloading the window, just refresh the record:
```suggestion
            "tag": "soft_reload",
```

Reply:
thanks